### PR TITLE
Remove refs to obsolete solver modules

### PR DESCRIPTION
**Pull request by peterschrammel**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: aa6fdbf44ddf442ea685ed5aef454e1ef8a48dd6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/94268615

**Review by kroening — APPROVED** [review]

**Review by smowton — APPROVED** [review]

File: jbmc/src/jbmc/jbmc_parse_options.h [hunk 1]
```diff
@@ -50,7 +50,7 @@ class optionst;
   "(classpath):(cp):(main-class):" \
   "(no-assertions)(no-assumptions)" \
   "(xml-ui)(json-ui)" \
-  "(smt1)(smt2)(fpa)(cvc3)(cvc4)(boolector)(yices)(z3)(opensmt)(mathsat)" \
+  "(smt1)(smt2)(fpa)(cvc3)(cvc4)(boolector)(yices)(z3)(mathsat)" \
```
Comment (thk123) [inline jbmc/src/jbmc/jbmc_parse_options.h:53]: :question: Do any docs need updating?

**Review by thk123 — APPROVED** [review]
Seems OK - could you remove the option from tg_parse_options too?

**Review by forejtv — APPROVED** [review]

File: jbmc/src/jbmc/jbmc_parse_options.h (hunk 1 above)
Comment (peterschrammel) [reply jbmc/src/jbmc/jbmc_parse_options.h:53]: I didn't find any mention of it.

**Pull request merged**